beacon-chain/state/state-native/getters_withdrawal_test.go

```diff
@@ -1,597 +1,353 @@
+				require.Equal(t, params.BeaconConfig().MaxWithdrawalsPerPayload, uint64(len(expected)))
+				withdrawal := &enginev1.Withdrawal{
+					Index:          0,
+					ValidatorIndex: 0,
+					Address:        vals[0].WithdrawalCredentials[12:],
+					Amount:         params.BeaconConfig().MaxEffectiveBalance + 1,
+				}
+				require.DeepEqual(t, withdrawal, expected[0])
+			})
+			t.Run("one fully withdrawable but zero balance", func(t *testing.T) {
+				s := state_native.EmptyStateFromVersion(t, stateVersion)
+				require.NoError(t, s.SetNextWithdrawalValidatorIndex(20))
+				vals := make([]*ethpb.Validator, 100)
+				balances := make([]uint64, 100)
+				for i := range vals {
+					balances[i] = params.BeaconConfig().MaxEffectiveBalance
+					val := &ethpb.Validator{
+						WithdrawalCredentials: make([]byte, 32),
+						EffectiveBalance:      params.BeaconConfig().MaxEffectiveBalance,
+						WithdrawableEpoch:     primitives.Epoch(1),
+					}
+					val.WithdrawalCredentials[0] = params.BeaconConfig().ETH1AddressWithdrawalPrefixByte
+					val.WithdrawalCredentials[31] = byte(i)
+					vals[i] = val
+				}
+				vals[3].WithdrawableEpoch = primitives.Epoch(0)
+				balances[3] = 0
+				require.NoError(t, s.SetValidators(vals))
+				require.NoError(t, s.SetBalances(balances))
+
+				expected, _, err := s.ExpectedWithdrawals()
+				require.NoError(t, err)
+				require.Equal(t, 0, len(expected))
+			})
+			t.Run("one partially withdrawable, one above sweep bound", func(t *testing.T) {
+				s := state_native.EmptyStateFromVersion(t, stateVersion)
+				vals := make([]*ethpb.Validator, 100)
+				balances := make([]uint64, 100)
+				for i := range vals {
+					balances[i] = params.BeaconConfig().MaxEffectiveBalance
+					val := &ethpb.Validator{
+						WithdrawalCredentials: make([]byte, 32),
+						EffectiveBalance:      params.BeaconConfig().MaxEffectiveBalance,
+						WithdrawableEpoch:     primitives.Epoch(1),
+					}
+					val.WithdrawalCredentials[0] = params.BeaconConfig().ETH1AddressWithdrawalPrefixByte
+					val.WithdrawalCredentials[31] = byte(i)
+					vals[i] = val
+				}
+				balances[3] += params.BeaconConfig().MinDepositAmount
+				balances[10] += params.BeaconConfig().MinDepositAmount
+				require.NoError(t, s.SetValidators(vals))
+				require.NoError(t, s.SetBalances(balances))
+				saved := params.BeaconConfig().MaxValidatorsPerWithdrawalsSweep
+				params.BeaconConfig().MaxValidatorsPerWithdrawalsSweep = 10
+				expected, _, err := s.ExpectedWithdrawals()
+				require.NoError(t, err)
+				require.Equal(t, 1, len(expected))
+				withdrawal := &enginev1.Withdrawal{
+					Index:          0,
+					ValidatorIndex: 3,
+					Address:        vals[3].WithdrawalCredentials[12:],
+					Amount:         params.BeaconConfig().MinDepositAmount,
+				}
+				require.DeepEqual(t, withdrawal, expected[0])
+				params.BeaconConfig().MaxValidatorsPerWithdrawalsSweep = saved
+			})
+		})
+	}
+
+	t.Run("electra all pending partial withdrawals", func(t *testing.T) {
```

Comment:
This test load a ssz test from the spec tests repo as a fully hydrated example rather than hand crafting test data.